src/rust/engine/fs/store/src/remote.rs

```diff
@@ -75,14 +75,20 @@ impl ByteStore {
             Level::Trace,
             desc = Some(format!("Storing {digest:?}")),
             |workunit| async move {
+                workunit.increment_counter(Metric::RemoteStoreWriteAttempts, 1);
                 let result = do_store().await;
 
-                if result.is_ok() {
-                    workunit.record_observation(
-                        ObservationMetric::RemoteStoreBlobBytesUploaded,
-                        digest.size_bytes as u64,
-                    );
-                }
+                let result_metric = match result {
+                    Ok(()) => {
+                        workunit.record_observation(
+                            ObservationMetric::RemoteStoreBlobBytesUploaded,
+                            digest.size_bytes as u64,
+                        );
+                        Metric::RemoteStoreWriteSuccesses
+                    }
+                    Err(_) => Metric::RemoteStoreWriteErrors,
+                };
+                workunit.increment_counter(result_metric, 1);
```

Comment:
Merging this single step into the two branches above would make the code flow easier, I think. And same for next file.

Reply:
As in, have two calls to `increment_counter`?

It might read a bit better, but my thinking was that it's nice to "guarantee" that we record the result against some metric, on every code path, which this style does better than having to remember to call `increment_counter` on every branch. Only a minor objection, though.